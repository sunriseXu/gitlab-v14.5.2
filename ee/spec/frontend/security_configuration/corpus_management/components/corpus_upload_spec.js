import { GlButton } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import CorpusUpload from 'ee/security_configuration/corpus_management/components/corpus_upload.vue';

const TEST_PROJECT_FULL_PATH = '/namespace/project';

describe('Corpus Upload', () => {
  let wrapper;

  const createComponentFactory = (mountFn = shallowMount) => (options = {}) => {
    const defaultProps = { totalSize: 4e8 };
    wrapper = mountFn(CorpusUpload, {
      propsData: defaultProps,
      mocks: {
        states: {
          uploadState: {
            progress: 0,
          },
        },
      },
      provide: {
        projectFullPath: TEST_PROJECT_FULL_PATH,
      },
      ...options,
    });
  };

  const createComponent = createComponentFactory();

  afterEach(() => {
    wrapper.destroy();
  });

  describe('component', () => {
    it('renders header', () => {
      createComponent();
      expect(wrapper.findComponent(GlButton).exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
