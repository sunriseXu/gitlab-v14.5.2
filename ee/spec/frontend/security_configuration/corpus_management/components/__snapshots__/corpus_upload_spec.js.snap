// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Corpus Upload component renders header 1`] = `
<div
  class="gl-h-11 gl-bg-gray-10 gl-display-flex gl-justify-content-space-between gl-align-items-center"
>
  <div
    class="gl-ml-5"
  >
    <gl-sprintf-stub
      message="Total Size: %{totalSize}"
    />
  </div>
   
  <gl-button-stub
    buttontextclasses=""
    category="primary"
    class="gl-mr-5"
    icon=""
    role="button"
    size="medium"
    tabindex="0"
    variant="confirm"
  >
    
    New corpus
  
  </gl-button-stub>
   
  <gl-modal-stub
    actioncancel="[object Object]"
    actionprimary="[object Object]"
    dismisslabel="Close"
    modalclass=""
    modalid="corpus-upload-modal"
    size="sm"
    title="New corpus"
    titletag="h4"
  >
    <corpus-upload-form-stub
      states="[object Object]"
    />
  </gl-modal-stub>
</div>
`;
